src/native/arc_list.rs

```diff
@@ -103,15 +103,21 @@ impl<T> Drop for ArcList<T> {
 }
 
 pub struct Node<T> {
-    next: AtomicUsize,
+    next: AtomicPtr<Node<T>>,
     enqueued: AtomicBool,
     data: T,
 }
 
 impl<T> Node<T> {
+    const EMPTY: *mut Node<T> = std::ptr::null_mut();
+
+    const fn sealed() -> *mut Node<T> {
+        std::ptr::null_mut::<Node<T>>().wrapping_add(1)
+    }
```

Comment:
It would be very cool if this were an associated constant, but `wrapping_add` isn't stably const yet, and I suspect this crate has some MSRV policy that would prevent just waiting a few weeks for it to be.